### PR TITLE
Refactor DataFusionInstanceLink usage

### PR DESCRIPTION
**Pull request by moiseenkov**
Refactored buggy `DataFusionPipelineLinkHelper` into a more generic function.

### CODE REVIEW

**Review by hussein-awala — APPROVED** [review]
Looks good

**Review by Lee-W — APPROVED** [review]

File: airflow/providers/google/cloud/operators/datafusion.py [hunk 1]
```diff
@@ -34,21 +34,12 @@
 from airflow.providers.google.cloud.operators.cloud_base import GoogleCloudBaseOperator
 from airflow.providers.google.cloud.triggers.datafusion import DataFusionStartPipelineTrigger
 from airflow.providers.google.cloud.utils.datafusion import DataFusionPipelineType
+from airflow.providers.google.cloud.utils.helpers import resource_path_to_dict
 
 if TYPE_CHECKING:
     from airflow.utils.context import Context
 
 
-class DataFusionPipelineLinkHelper:
```
Comment (eladkal) [inline airflow/providers/google/cloud/operators/datafusion.py:42]: is this class part of the public API?
Do we need to deprecate it first?

Comment (moiseenkov) [reply airflow/providers/google/cloud/operators/datafusion.py:42]: You are right. Updated PR.

**Comment by VladaZakharova** [comment]
@eladkal @potiuk 
Hi Team! Can you please take a look on the changes once again? Thanks!

File: airflow/providers/google/cloud/operators/datafusion.py [hunk 2]
```diff
@@ -34,16 +35,25 @@
...
+    Helper class for Pipeline links.
+
+    .. warning::
+        This class is deprecated. Consider using ``resource_path_to_dict()`` instead.
+    """
 
     @staticmethod
     def get_project_id(instance):
+        warnings.warn(
+            "DataFusionPipelineLinkHelper is deprecated. Consider using resource_path_to_dict() instead."
```
Comment (eladkal) [inline airflow/providers/google/cloud/operators/datafusion.py]: You need to raise `AirflowProviderDeprecationWarning`
also the word consider suggests there is a choice... it's not the case. users must migrate as the class will be removed in future release

Comment (moiseenkov) [reply airflow/providers/google/cloud/operators/datafusion.py]: Thanks for the comment. Fixed. Please re-review.

**Review by eladkal — APPROVED** [review]

**Pull request merged**